### PR TITLE
CSSRE-1853: add get-node-restart script

### PR DESCRIPTION
**Pull request by vinaybommana**
adding a script to check the nodes restarted in the last X minutes and list the pods associated with the respective node

### CODE REVIEW

**Comment by reetika-vyas** [comment]
CI is failing with below error

```


13:02:35 [openshift-managed-scripts-gh-pr-check] $ /bin/sh -xe /tmp/jenkins8437045871171599477.sh
13:02:35 + ./hack/app_sre_pr_check.sh
13:02:35 ./hack/schema_validation.sh
13:02:35 convert ./scripts/CEE/check-tech-preview-features/metadata.yaml to json
13:02:35 convert succeed
13:02:35 convert ./scripts/CEE/etcd-health-check/metadata.yaml to json
13:02:35 convert succeed
13:02:35 convert ./scripts/CEE/get-kafka-instance-state/metadata.yaml to json
13:02:35 convert succeed
13:02:35 convert ./scripts/CEE/get-rhosak-operators/metadata.yaml to json
13:02:36 convert succeed
13:02:36 convert ./scripts/CEE/list-alerts/metadata.yaml to json
13:02:36 convert succeed
13:02:36 convert ./scripts/CEE/registry-s3-bucket-size/metadata.yaml to json
13:02:36 convert succeed
13:02:36 convert ./scripts/SREP/cluster-operator-status/metadata.yaml to json
13:02:36 convert succeed
13:02:36 convert ./scripts/SREP/delete-olm-operator/metadata.yaml to json
13:02:36 convert succeed
13:02:36 convert ./scripts/SREP/elasticsearch-status/metadata.yaml to json
13:02:36 convert succeed
13:02:36 convert ./scripts/SREP/example/metadata.yaml to json
13:02:36 convert succeed
13:02:36 convert ./scripts/SREP/get-targets-down/metadata.yaml to json
13:02:36 convert succeed
13:02:36 convert ./scripts/SREP/manage-silence/metadata.yaml to json
13:02:36 convert succeed
13:02:36 convert ./scripts/SREP/node-logs/metadata.yaml to json
13:02:36 convert succeed
13:02:36 convert ./scripts/SREP/tsdb-status/metadata.yaml to json
13:02:36 convert succeed
13:02:36 convert ./scripts/CSSRE/get-node-restarts/metadata.yaml to json
13:02:36 convert succeed
13:02:36 validating the jsonschema for ./scripts/CEE/check-tech-preview-features/metadata.json
13:02:36 Unable to find image 'quay.io/haowang/jsonschema:latest' locally
13:02:36 latest: Pulling from haowang/jsonschema
13:02:36 257263888892: Pulling fs layer
13:02:36 acd29ab1ce45: Pulling fs layer
13:02:36 640bc92e03f8: Pulling fs layer
13:02:36 0f7f66ce0910: Pulling fs layer
13:02:36 40b8673e5b5c: Pulling fs layer
13:02:36 34bfcc984998: Pulling fs layer
13:02:36 538c5948b21f: Pulling fs layer
13:02:36 81cb975c6e51: Pulling fs layer
13:02:36 42034767b6ff: Pulling fs layer
13:02:36 7bff1891cbe9: Pulling fs layer
13:02:36 0f7f66ce0910: Waiting
13:02:36 40b8673e5b5c: Waiting
13:02:36 34bfcc984998: Waiting
13:02:36 538c5948b21f: Waiting
13:02:36 81cb975c6e51: Waiting
13:02:36 42034767b6ff: Waiting
13:02:36 7bff1891cbe9: Waiting
13:02:36 acd29ab1ce45: Verifying Checksum
13:02:36 acd29ab1ce45: Download complete
13:02:36 257263888892: Verifying Checksum
13:02:36 257263888892: Download complete
13:02:37 0f7f66ce0910: Verifying Checksum
13:02:37 0f7f66ce0910: Download complete
13:02:37 640bc92e03f8: Verifying Checksum
13:02:37 640bc92e03f8: Download complete
13:02:37 538c5948b21f: Verifying Checksum
13:02:37 538c5948b21f: Download complete
13:02:37 34bfcc984998: Verifying Checksum
13:02:37 34bfcc984998: Download complete
13:02:37 40b8673e5b5c: Verifying Checksum
13:02:37 40b8673e5b5c: Download complete
13:02:37 257263888892: Pull complete
13:02:37 81cb975c6e51: Verifying Checksum
13:02:37 81cb975c6e51: Download complete
13:02:37 7bff1891cbe9: Verifying Checksum
13:02:37 7bff1891cbe9: Download complete
13:02:37 42034767b6ff: Verifying Checksum
13:02:37 42034767b6ff: Download complete
13:02:37 acd29ab1ce45: Pull complete
13:02:38 640bc92e03f8: Pull complete
13:02:38 0f7f66ce0910: Pull complete
13:02:38 40b8673e5b5c: Pull complete
13:02:38 34bfcc984998: Pull complete
13:02:38 538c5948b21f: Pull complete
13:02:38 81cb975c6e51: Pull complete
13:02:39 42034767b6ff: Pull complete
13:02:39 7bff1891cbe9: Pull complete
13:02:39 Digest: sha256:edddfa33a567a464e1f35b4ba6acb908da1272c07670267283b0b8d359f07b62
13:02:39 Status: Downloaded newer image for quay.io/haowang/jsonschema:latest
13:02:40 validation succeed
13:02:40 validating the jsonschema for ./scripts/CEE/etcd-health-check/metadata.json
13:02:41 validation succeed
13:02:41 validating the jsonschema for ./scripts/CEE/get-kafka-instance-state/metadata.json
13:02:43 validation succeed
13:02:43 validating the jsonschema for ./scripts/CEE/get-rhosak-operators/metadata.json
13:02:44 validation succeed
13:02:44 validating the jsonschema for ./scripts/CEE/list-alerts/metadata.json
13:02:45 validation succeed
13:02:45 validating the jsonschema for ./scripts/CEE/registry-s3-bucket-size/metadata.json
13:02:46 validation succeed
13:02:46 validating the jsonschema for ./scripts/SREP/cluster-operator-status/metadata.json
13:02:47 validation succeed
13:02:47 validating the jsonschema for ./scripts/SREP/delete-olm-operator/metadata.json
13:02:49 validation succeed
13:02:49 validating the jsonschema for ./scripts/SREP/elasticsearch-status/metadata.json
13:02:50 validation succeed
13:02:50 validating the jsonschema for ./scripts/SREP/example/metadata.json
13:02:51 validation succeed
13:02:51 validating the jsonschema for ./scripts/SREP/get-targets-down/metadata.json
13:02:53 validation succeed
13:02:53 validating the jsonschema for ./scripts/SREP/manage-silence/metadata.json
13:02:54 validation succeed
13:02:54 validating the jsonschema for ./scripts/SREP/node-logs/metadata.json
13:02:55 validation succeed
13:02:55 validating the jsonschema for ./scripts/SREP/tsdb-status/metadata.json
13:02:56 validation succeed
13:02:56 validating the jsonschema for ./scripts/CSSRE/get-node-restarts/metadata.json
13:02:57 Python: 'Python' is not one of ['python', 'bash']
13:02:57 validation failed: ./scripts/CSSRE/get-node-restarts/metadata.json
13:02:57 make: *** [validation] Error 1
13:02:57 Build step 'Execute shell' marked build as failure
13:02:58 Setting status of cd01b72190c100b19405349c91d2ddbd06de89fa to FAILURE with url https://ci.ext.devshift.net/job/openshift-managed-scripts-gh-pr-check/247/ and message: ' '
13:02:58 Using context: ci.ext.devshift.net PR build
13:02:58 Finished: FAILURE
```

**Comment by T0MASD** [comment]
There are many files in [scripts/CSSRE/lib/py/__pycache__/](https://github.com/openshift/managed-scripts/pull/57/files#diff-577fdf84726992e77aff63e1fa5c4447b6f338676b44993d17743d0e868b3a56) dir, they are not nesacery to avoid this in the future  `__pycache__` and `*.pyc` should be added to `.gitignore`

**Comment by georgettica** [comment]
please remove the bad files in addition to adding a gitignore (`scripts/CSSRE/lib/py/__pycache__/`)

File: Dockerfile [hunk 1]
```diff
@@ -52,6 +52,9 @@ COPY --from=build-stage0 /aws/bin/  /usr/local/bin
 COPY --from=build-stage0 /usr/local/aws-cli /usr/local/aws-cli
 COPY scripts /managed-scripts
 
+# Install python packages
+RUN python3 -m pip install -r /managed-scripts/CSSRE/requirements.txt --user
```
Comment (georgettica) [inline Dockerfile:56]: seems to be awfully specific file ..
should these packages be hardcoded into the Dockerfile so all will be able to use them?

this is done so changes in  `/managed-scripts/CSSRE/requirements.txt` won't cause voodoo and add stuff to our container without us recognizing it

Comment (pepedocs) [reply Dockerfile:56]: @georgettica 
> should these packages be hardcoded into the Dockerfile so all will be able to use them?
This was not the intention. And the upside of using the file is that it is easy to tell which team provided these packages, secondly, we don't have to build a container to run a simple python script in our local machine. IMO putting it at the top-level doesn't guarantee from someone adding "voodoo", everything goes to a PR anyway. 

However if putting every package in the dockerfile is the pattern we want to follow, then we can update this.

Comment (vinaybommana) [reply Dockerfile]: @georgettica modified the `Dockerfile` with required packages, rather than using `requirements.txt` file

**Comment by fahlmant** [comment]
Please add a description

File: Dockerfile (hunk 1 above)
Comment (T0MASD) [reply Dockerfile]: +1 for adding requests and tabulate python packages. 
My preference via `dnf install python3-tabulate.noarch python3-requests` but I guess `pip3 install requests tabulate` would work too.

**Comment by vinaybommana** [comment]
@fahlmant just added the description. thanks

**Comment by reetika-vyas** [comment]
Failed with below error

```
10:39:30 /usr/bin/docker pull "registry.access.redhat.com/ubi8/python-36:latest"
10:39:30 latest: Pulling from ubi8/python-36
10:39:30 Digest: sha256:b3f3ebe3cd793144964c550dc2b70aeb5d777c9f04dec7b9bda0abee0f595194
10:39:30 Status: Image is up to date for registry.access.redhat.com/ubi8/python-36:latest
10:39:30 /usr/bin/docker run -v /srv/volume/jenkins/workspace/openshift-managed-scripts-gh-pr-check:/app --entrypoint=/bin/sh -w=/app/scripts "registry.access.redhat.com/ubi8/python-36:latest" -c "pip3 install pyflakes && find . -name '*.py' -print0 | xargs -0 -n1 pyflakes "
10:39:31 Collecting pyflakes
10:39:31   Downloading pyflakes-2.4.0-py2.py3-none-any.whl (69 kB)
10:39:31 Installing collected packages: pyflakes
10:39:31 Successfully installed pyflakes-2.4.0
10:39:31 WARNING: You are using pip version 21.0.1; however, version 21.3.1 is available.
10:39:31 You should consider upgrading via the '/opt/app-root/bin/python3.6 -m pip install --upgrade pip' command.
10:39:31 ./CSSRE/get-node-restarts/script.py:11:1 'os' imported but unused
10:39:32 make: *** [pyflakes] Error 123
10:39:32 Build step 'Execute shell' marked build as failure
10:39:32 Setting status of fa52d61c61255e959969b842b558577ea985f768 to FAILURE with url https://ci.ext.devshift.net/job/openshift-managed-scripts-gh-pr-check/255/ and message: ' '
10:39:32 Using context: ci.ext.devshift.net PR build
10:39:32 Finished: FAILURE
```

**Comment by vinaybommana** [comment]
Hi @wanghaoran1988 please close this merge request.. I'll raise a new PR with proper details

**Pull request closed without merging**